src/Symfony/Component/Security/Http/Controller/UserValueResolver.php

```diff
@@ -35,12 +35,8 @@ public function __construct(TokenStorageInterface $tokenStorage)
 
     public function supports(Request $request, ArgumentMetadata $argument): bool
     {
-        if ($argument->getAttribute() instanceof CurrentUser) {
-            return true;
-        }
-
         // only security user implementations are supported
```

Comment:
This comment is not 100% true anymore

Reply:
Could you elaborate? It looks right to me given that only UserInterface instances are supported

Reply:
What this originally meant was: The method needs to have a typehint for `UserInterface` (rather than e.g. `ConcreteAppUser`). This changes with the `#[CurrentUser]` attribute, which allows the typehint to be `ConcreteAppUser`.

Maybe "// only security user implementations are supported without the CurrentUser attribute" or something covers the conditions more precisely?

Reply:
Got it, thanks. Comment updated